pages/_app.js

```diff
@@ -46,15 +61,41 @@ export default function App({ Component, pageProps }) {
     router.push("/");
   }
 
-  function handleFilterValue(value) {
-    setFilter(value);
-    setFilterCount("1");
+  function handleFilterValue(event) {
+    if (event.name === "fertiliserSeason") {
+      setFilters((prevFilters) => {
+        const updatedFertiliserSeason = prevFilters.fertiliserSeason.includes(
+          event.value
+        )
+          ? prevFilters.fertiliserSeason.filter(
+              (season) => season !== event.value
+            )
+          : [...prevFilters.fertiliserSeason, event.value];
+        return {
+          ...prevFilters,
+          fertiliserSeason: updatedFertiliserSeason,
+        };
+      });
+    } else {
+      setFilters((prevFilters) => ({
+        ...prevFilters,
+        [event.name]: event.value,
+      }));
+    }
+  }
```

Comment:
This function has complex conditional logic for handling multiple selections in `fertiliserSeason` - I know who the fan of complex conditional logic is 😁 

_But I want to mention a simpler way_

➡️ Create a utility function to handle array-based filters.

Example:
```js

function updateArrayFilter(currentValues, value) {
  return currentValues.includes(value)
 // Remove value if it exists
    ? currentValues.filter((_value) => _value !== value) 
 // Add value if it doesn't exist
    : [...currentValues, value];
}

function handleFilterValue(event) {
  const { name, value } = event;

  setFilters((prevFilters) => {
    if (name === "fertiliserSeason") {
      return {
        ...prevFilters,
        fertiliserSeason: updateArrayFilter(prevFilters.fertiliserSeason, value),
      };
    }

    return {
      ...prevFilters,
      [name]: value,
    };
  });
}
```